packages/curve-ui-kit/src/shared/ui/Accordion.tsx

```diff
@@ -0,0 +1,170 @@
+ * single-item constrained collapse component. It's called an Accordion in our Figma
+ * design system for consistency. We created this custom implementation because the
+ * MUI Accordion lacks the customization options needed for our design requirements,
+ * particularly for features like the ghost variant and responsive sizing with
+ * appropriate effects on children elements.
+ */
+export const Accordion = ({
+  title,
+  icon,
+  ghost = false,
+  size = 'small',
+  info,
+  defaultExpanded = false,
+  children,
+}: Props) => {
+  const [isOpen, , , toggle] = useSwitch(defaultExpanded)
+  const id = `accordion-${useId()}`
+
+  return (
+    <Stack>
+      <ButtonBase
+        onClick={toggle}
+        aria-expanded={isOpen}
+        aria-controls={id}
+        sx={{
+          paddingBlock: Spacing.sm,
+          paddingInline: ghost ? 0 : Spacing.sm,
+          ...(isOpen && !ghost && { backgroundColor: layer1Fill }),
+          transition: `background-color ${TransitionFunction}`,
+
+          // Render border insize without layout shift on hover
+          position: 'relative',
+          '&::after': {
+            content: '""',
+            position: 'absolute',
+            inset: 0,
+            ...(ghost ? { borderBottom: borderStyle } : { border: borderStyle }),
+          },
+
+          '&:hover, &.Mui-focusVisible': {
+            '&::after': {
+              borderColor: (t) => t.design.Button.Focus_Outline,
+              borderWidth: '2px',
+            },
+            ...(!ghost && { backgroundColor: layer1Fill }),
+          },
+        }}
+      >
+        <Stack flexGrow={1} direction="row" alignItems="center" gap={Spacing.sm}>
+          {icon && (
+            <Box
+              display="flex"
+              justifyContent="center"
+              alignItems="center"
+              sx={{
+                width: IconSize.md,
+                height: IconSize.md,
+              }}
+            >
+              {icon}
+            </Box>
+          )}
+
+          {typeof title === 'string' ? (
+            <Typography
+              flexGrow={1}
+              variant={titleVariants[size]}
+              color="textSecondary"
+              sx={{
+                textAlign: 'start',
+              }}
+            >
+              {title}
+            </Typography>
+          ) : (
+            <Box display="flex" flexGrow={1}>
+              {title}
+            </Box>
+          )}
+
+          {info}
+
+          <ExpandMoreIcon
+            sx={{
+              width: IconSize.md,
+              height: IconSize.md,
+              transform: isOpen ? 'rotate(180deg)' : 'rotate(0deg)',
+              // Create a transition that mimics the collapse transition
+              transition: (t) =>
+                t.transitions.create('transform', {
+                  duration: t.transitions.duration.standard,
+                  easing: t.transitions.easing.easeInOut,
+                }),
```

Comment:
Isn't this the same as 
```suggestion
              transition: `transform ${TransitionFunction}`
```
?

Reply:
No, that is our own made up transition function. I specifically wanted the transition time and easing that is used by the collapse component. maybe we should use that as well instead of using our own custom TransitionFunction? 🤔

Reply:
That function (`ease-out 256ms`) comes from @0xtutti's design system.
Actually we should customize the MUI theme to use that, and change the `TransitionFunction` usages to `theme.transitions.create([...])`